core/src/test/java/org/elasticsearch/common/geo/GeoWKTShapeParserTests.java

```diff
@@ -239,5 +240,26 @@ public void testInvalidGeometryType() throws IOException {
         assertValidException(builder, IllegalArgumentException.class);
     }
 
-    // todo: add geocollection
+    @Override
+    public void testParseGeometryCollection() throws IOException {
+        PolygonBuilder polygonBuilder = PolygonBuilder.class.cast(
+            RandomShapeGenerator.createShape(random(), RandomShapeGenerator.ShapeType.POLYGON));
+        List<Coordinate> lineStringCoords = randomLineStringCoords();
+        LineStringBuilder lineStringBuilder = new LineStringBuilder(lineStringCoords);
+        List<Coordinate> multiPointCoords = randomLineStringCoords();
+        MultiPointBuilder multiPointBuilder = new MultiPointBuilder(multiPointCoords);
+        GeometryCollectionBuilder builder = new GeometryCollectionBuilder()
+            .shape(polygonBuilder).shape(lineStringBuilder).shape(multiPointBuilder);
```

Comment:
Can this be randomized so the GeometryCollectionBuilder has between 0 and e.g. 2 or 3 shapes?